thunder/dynamo/compiler.py

```diff
@@ -12,6 +30,262 @@ def _warn_thunder_compiler():
+
+        target = node.target  # Target is the function to call.
+        if node.op == "call_method":
+            self_arg = node.args[0]
+            target = getattr(torch.Tensor, node.target, None)
+            assert target is not None, f"Failed to find method {node.target}"
+
+        # If the operation has automatic registration, we mark it as unsupported as `inductor` might be
+        # able to deal with it better.
+        if target in auto_register_ops:
+            self.split_reasons.append(
+                SplitReason(
+                    SplitReasonType.MISSING_OP_SUPPORT,
+                    info=f"node with name: {node.name} and target: {node.target} only has an automatic torch fallback in thunder.",
+                )
+            )
+            return False
+
+        # If thunder has a mapping for this operation, try executing the meta function and see.
+        # We have a symbol for `torch.where`, but we don't support one overload of it.
+        # So, we try and execute the meta to get a real signal.
+        #
+        # Regarding `inspect.isbuiltin`, dynamo graph uses `+`, `>` which are builtin `add`, `gt`.
+        # We try to proxify the arguments and call these operations on them to see if they are supported.
+        if target in _torch_to_thunder_function_map or inspect.isbuiltin(target):
+            if target in [torch.ones]:  # Factory functions. (removing this will lead to split)
+                # NOTE - Factory functions don't work as the expect `_cache_info` to be populated
+                #        with default dtype but `_cache_info` is only created and populated in `thunder.jit` path.
+                # my_trc = TraceCtx()
+                # with tracectx(my_trc):
+                #     thunder.torch.ones(3, 3)
+                return True
+
+            thunder_symbol_or_builtin = _torch_to_thunder_function_map.get(target, target)
+            did_run, opt_split_reason = try_execute_symbol(thunder_symbol_or_builtin, node)
+            if opt_split_reason is not None:
+                self.split_reasons.append(opt_split_reason)
+            return did_run
+
+        # We found no automatic fallback registration and no mapping to thunder symbol.
+        self.split_reasons.append(
+            SplitReason(
+                SplitReasonType.MISSING_OP_SUPPORT,
+                info=f"node with name: {node.name} and target: {node.target} didn't have any mapping in thunder.",
+            )
+        )
+        return False
+
+
+def _all_graph_supported_by_thunder(gm: torch.fx.GraphModule, sample_input: list[torch.SymInt, torch.Tensor]) -> bool:
+    """
+    Determine whether there is any thunder unsupported operation.
+    """
+    # NOTE - Unused for now.
+    op_support = ThunderOperatorSupport(gm)
+    supported = True
+    for node in gm.graph.nodes:
+        if node.op in ["call_method", "call_function"]:
+            supported = op_support.is_node_supported(gm, node)
```

Comment:
Is `ThunderOperatorSupport` really needed? Can it be simply a function `is_node_supported_by_thunder`?